files/startup-script-stdlib.sh

```diff
@@ -0,0 +1,230 @@
+#! /bin/bash
+#
+# Copyright 2018 Google LLC. This software is provided as-is,
+# without warranty or representation for any use or purpose.
+# Your use of it is subject to your agreements with Google.
+
+# Standard library of functions useful for startup scripts.
+
+# Pending UX behaviors
+# TODO(jmccune): load_config_values to load user configuration
+# TODO(jmccune): call mandatory_argument() before E_MISSING_MANDATORY_ARG
+# TODO(jmccune): add -c <name> for alternate startup-script-custom name
+# Pending initialization functions:
+# TODO(jmccune): gsutil initialization w/ crcmod
+# Pending operational functions:
+# TODO(jmccune): get_from_bucket()
+# TODO(jmccune): setup_init_script()
+# TODO(jmccune): setup_sudoers()
+
+# These are outside init_global_vars so logging functions work with the most
+# basic case of `source startup-script-stdlib.sh`
+readonly SYSLOG_DEBUG_PRIORITY="${SYSLOG_DEBUG_PRIORITY:-syslog.debug}"
+readonly SYSLOG_INFO_PRIORITY="${SYSLOG_INFO_PRIORITY:-syslog.info}"
+readonly SYSLOG_ERROR_PRIORITY="${SYSLOG_ERROR_PRIORITY:-syslog.error}"
+# Global counter of how many times stdlib::init() has been called.
+STARTUP_SCRIPT_STDLIB_INITIALIZED=0
+
+# Error codes
+readonly E_RUN_OR_DIE=5
+readonly E_MISSING_MANDATORY_ARG=9
+readonly E_UNKNOWN_ARG=10
+
+stdlib::debug() {
+  [[ -z "${DEBUG:-}" ]] && return 0
+  local ds msg="$*"
```

Comment:
I've had shellcheck squawk at me when I declared and assigned to a local variable in the same expression. The fact that we're declaring `ds` as a local, and then declaring `msg` as a local and then assigning it is a little surprising. In terms of correctness this code is fine as-is, but I think it might be more readable to break out `ds` into a separate declaration. Since this is a style comment, feel free to change if you agree, and resolve this comment otherwise.

Reply:
I went back and forth on this one too.  Where I settled with this code is, "make shellcheck happy and move on."

FWIW, shellcheck only seems to complain when the assignment happens in a subshell.  e.g. it'll complain about `local ds="$(date +%s)"` but it won't complain about `local msg="$*"`.

I'll go ahead and break these out since the shellguide mentions it as well.

Reply:
Fixed